### ISSUE TITLE
[BUG] Styles won't apply to local websites behind reverse proxy

### ISSUE THREAD

**Issue by EmTeeAge** [darkreader/darkreader#3379]
### **Describe the bug**

The toggle in Dark Reader is stuck disabled on local websites accessed through a local domain. When I try to go to a local website like something.example.com, the button to toggle Dark Reader on is simply not working, no matter how many times I click it. However, when I try to access the same website using IP like http://192.168.1.160:portnumber and bypass my Nginx reverse proxy, the Dark Reader dark mode gets applied to the website properly.

To be clear about my Nginx reverse proxy - the reverse proxy is sitting in front of my webserver and catches all 80 and 443 requests and redirects them to the webserver behind it. For some reason, something in this proccess makes Dark Reader unable to apply styles. I don't really understand what's going on, since most normal websites use reverse proxies as well and without any issues.

### **Steps to reproduce**

Well, this is kind of hard to reproduce since you would have to have something behind Nginx reverse proxy on a local network to try it.

### **Expected behavior**

Styling should apply to my local website just like to any other website on the Internet.

### **Actual behavior**

Connecting to a local web server using it's IP and port --> styles applied successfully
Connecting to a local web server using HTTPS and domain name --> styles won't apply, button is stuck disabled
Connecting to a local web server using HTTP and domain name --> styles won't apply button is stuck disabled

### **Screenshots**
Using IP:
![image](https://user-images.githubusercontent.com/67438185/89835513-754df300-db54-11ea-8f39-18b6e7be6cb0.png)
Using HTTPS and domain:
![image](https://user-images.githubusercontent.com/67438185/89835522-77b04d00-db54-11ea-80f3-9bf99ee280b6.png)
Using HTTP and domain:
![image](https://user-images.githubusercontent.com/67438185/89835600-91ea2b00-db54-11ea-94cd-8e297b211b43.png)
The button won't toggle on
![image](https://user-images.githubusercontent.com/67438185/89835661-aa5a4580-db54-11ea-9cbb-9341a3b6ee40.png)

### **System Information:**
 - OS: Windows 10
 - Browser: Firefox 79, ungoogle Chromium 81 - issue persistent across both browsers
 - Dark Reader version: 4.9.16

### **Additional context**

I'm sorry if this is not the right place to ask, since I'm suspecting the problem to be related to Nginx, but I was curious how could that happen and what exactly would prevent Dark Reader from applying dark mode. It looks kind of like it's treated as a browser-protected website, similarly to how it's not possible to apply styling to Chrome Web Store or Settings, etc.

**Comment by Gusted** [comment]
Any errors in console or warnings?

**Comment by EmTeeAge** [comment]
Mhh. I only tried checking the Network tab, but the requests are essentially the same, in both cases making 43 successful requests to the server:
![image](https://user-images.githubusercontent.com/67438185/89836384-2c973980-db56-11ea-9ff6-b6975c58c9f9.png)

However, looking at the console, there seems to be a difference:
![image](https://user-images.githubusercontent.com/67438185/89836412-3d47af80-db56-11ea-8974-b0e0618bacb1.png)

Although in case of Chromium, the console output is the same:
![image](https://user-images.githubusercontent.com/67438185/89836504-6d8f4e00-db56-11ea-843f-5ab891a72291.png)

All of that, however seems to be related to Urbackup itself, because when I tried other local websites, for example my music streaming webapp, there seems to be nothing in the console, but styles still won't apply:
![image](https://user-images.githubusercontent.com/67438185/89836771-fdcd9300-db56-11ea-85f1-db775b859490.png)

**Comment by Gusted** [comment]
What if you enable/disable/enable

**Comment by EmTeeAge** [comment]
Repeatedly clicking the toggle button does nothing, the cross inside of that button never changes to a checkmark. Turning the global toggle On/Off also doesn't do anything to the website, except for changing my browser theme.

![image](https://user-images.githubusercontent.com/67438185/89837152-dcb97200-db57-11ea-8370-92b040639837.png)
![image](https://user-images.githubusercontent.com/67438185/89837185-ed69e800-db57-11ea-8d14-c686142d41cc.png)

**Comment by Gusted** [comment]
How does your sitelist looks like?

**Comment by EmTeeAge** [comment]
Oh, so I checked my Site List and my domains were in the "Not Invert listed" list, which I found weird, since I though clicking the toggle would move websites from "Invert" to "Not Invert" and vice versa.
![image](https://user-images.githubusercontent.com/67438185/89886468-2f2c7a00-dbbc-11ea-878a-5aaf1f753c88.png)

When I deleted the entries from that list, the styles applied correctly, however they also applied to my main domain "doma.io", which I didn't want to. That's where I noticed a strange behavior - when I toggled dark mode off for the main domain "doma.io" it added an entry for that domain to "Not Invert listed" list. However, doing that also removed dark mode from all subdomains like "example.doma.io" even though they weren't listed. When I went to any subdomain and clicked the toggle, it just added/removed entry from the list, but the site remained white, without any dark mode applied.

What I think is happening is that if you turn off dark mode for the main domain, it turns off dark mode for all subdomain as well. At the same time, once the dark mode is off for the main domain, you are unable to switch it on for individual subdomains and the toggle appears to do nothing, except for adding/removing the subdomain from the "Not Invert listed" list. Once you switch dark mode on for the main domain, you are able to individually controll dark mode on/off for the subdomains.

In simplier words:
Dark mode ON on main domain --> being able to individually control dark mode on subdomains
Dark mode OFF on main domain --> not being able to individually control dark mode on subdomains

Made a quick screen capture to show the behavior here: https://www.youtube.com/watch?v=bZgl0NPeTG4&feature=youtu.be
